RecoPixelVertexing/PixelLowPtUtilities/src/TrackCleaner.cc

```diff
@@ -53,306 +47,269 @@ class HitComparatorByRadius
-              << " keep lower d0 " << tracks[i].first->d0()
-                            << " " << tracks[j].first->d0()
-              << ", keep #" << (keep[i] ? i : ( keep[j] ? j : 9999 ) );
-  
-            changes++;
-          }
-        }
-        else
-        { // note more than 50%, but at least two are shared
-          if((*sharing).second > 1)
-          {
-            if(tracks[i].second.size() != tracks[j].second.size())
-            { // keep longer
-              if(tracks[i].second.size() > tracks[j].second.size()) 
-                keep[j] = false; else keep[i] = false; 
+
               changes++;
+            } else {  // there is a common layer, keep smaller impact
+              if (fabs(tracks[i].first->d0()) < fabs(tracks[j].first->d0()))
+                keep[j] = false;
+              else
+                keep[i] = false;
+
+              LogTrace("TrackCleaner") << "   Clash #" << i << " #" << j << " keep lower d0 " << tracks[i].first->d0()
+                                       << " " << tracks[j].first->d0() << ", keep #"
+                                       << (keep[i] ? i : (keep[j] ? j : 9999));
 
-              LogTrace("TrackCleaner")
-                << "   Sharing " << (*sharing).second << " remove by size";
-            }
-            else
-            { // keep smaller impact
-              if(fabs(tracks[i].first->d0())
-               < fabs(tracks[j].first->d0()))
-                keep[j] = false; else keep[i] = false; 
               changes++;
+            }
+          } else {  // note more than 50%, but at least two are shared
+            if ((*sharing).second > 1) {
+              if (tracks[i].second.size() != tracks[j].second.size()) {  // keep longer
+                if (tracks[i].second.size() > tracks[j].second.size())
+                  keep[j] = false;
+                else
+                  keep[i] = false;
+                changes++;
+
+                LogTrace("TrackCleaner") << "   Sharing " << (*sharing).second << " remove by size";
+              } else {  // keep smaller impact
+                if (fabs(tracks[i].first->d0()) < fabs(tracks[j].first->d0()))
+                  keep[j] = false;
+                else
+                  keep[i] = false;
+                changes++;
+
+                LogTrace("TrackCleaner") << "   Sharing " << (*sharing).second << " remove by d0";
+              }
+            }
+          }
+        } else {  // pair tracks
+          if ((*sharing).second > 0) {
+            // Remove second track
+            keep[j] = false;
 
-              LogTrace("TrackCleaner")
-                << "   Sharing " << (*sharing).second << " remove by d0";
-            } 
+            changes++;
           }
         }
       }
-      else
-      { // pair tracks
-        if((*sharing).second > 0)
-        {
-          // Remove second track
-          keep[j] = false;
-  
-          changes++;
-        }
-      }
-    }
-/*
+      /*
```

Comment:
This could be probably changed in the rules (IMHO): for comments spanning several lines with
```C++
/*
...
*/
```

I think it is most convenient (for clarity) signalling the commented out code with a ```\*``` at the beginning of the line